codec/decoder/core/src/decoder.cpp

```diff
@@ -372,6 +374,33 @@ void WelsDecoderLastDecPicInfoDefaults (SWelsLastDecPicInfo& sLastDecPicInfo) {
   sLastDecPicInfo.bLastHasMmco5 = false;
 }
 
+/*!
+* \brief   copy SpsPps from one Ctx to another ctx for threaded code
+*/
+void CopySpsPps (PWelsDecoderContext pFromCtx, PWelsDecoderContext pToCtx) {
+  pToCtx->sSpsPpsCtx = pFromCtx->sSpsPpsCtx;
+  PAccessUnit pFromCurAu = pFromCtx->pAccessUnitList;
+  PSps pTmpLayerSps[MAX_LAYER_NUM];
+  for (int i = 0; i < MAX_LAYER_NUM; i++) {
+    pTmpLayerSps[i] = NULL;
+  }
+  // track the layer sps for the current au
+  for (unsigned int i = pFromCurAu->uiStartPos; i <= pFromCurAu->uiEndPos; i++) {
+    uint32_t uiDid = pFromCurAu->pNalUnitsList[i]->sNalHeaderExt.uiDependencyId;
+    pTmpLayerSps[uiDid] = pFromCurAu->pNalUnitsList[i]->sNalData.sVclNal.sSliceHeaderExt.sSliceHeader.pSps;
+    for (unsigned int j = 0; j < MAX_SPS_COUNT + 1; ++j) {
```

Comment:
You are right. It should be "from". However, pToCtx->sSpsPpsCtx = pFromCtx->sSpsPpsCtx has already called. So pToCtx->sSpsPpsCtx.sSpsBuffer and pFromCtx->sSpsPpsCtx.sSpsBuffer are identical.

Reply:
There is bPpsAvailFlags[] in the data structure

Reply:
bPpsAvailFlags[] does not seem to be used anywhere as neccessay info. It may not need to include it. However it is not harmful, I guess.

Reply:
OK. Please modify "to" to "from" for correct logical understanding.

Reply:
Actually it turns out I am right. it is intended to be "to" and must be "to" because pToCtx->sSpsPpsCtx.pActiveLayerSps is a pointer which must point to the copied pToCtx->sSpsPpsCtx.sSpsBuffer since assignment "pToCtx->sSpsPpsCtx = pFromCtx->sSpsPpsCtx;"
So I am very sure that current implementation is correct.